src/util/arith_tools.cpp

```diff
@@ -285,11 +287,33 @@ bool get_bvrep_bit(
   std::size_t width,
   std::size_t bit_index)
 {
-  // The representation is binary, using '0'/'1',
-  // most significant bit first.
   PRECONDITION(bit_index < width);
-  PRECONDITION(src.size() == width);
-  return src[src.size() - 1 - bit_index] == '1';
+
+  // The representation is hex, most significant nibble first.
+  const auto nibble_index = bit_index >> 2;
```

Comment:
Maybe leave the optimisation that "divide by 4" is a shift by 2 to the compiler? I don't think doing those optimisations help readability.